app/models/manageiq/providers/google/inventory/parser/network_manager.rb

```diff
@@ -0,0 +1,352 @@
+          :source_security_group => nil
+        )
+      end
+    end
+  end
+
+  def load_balancers
+    forwarding_rules = collector.forwarding_rules
+
+    forwarding_rules.each do |forwarding_rule|
+      persister_load_balancer = persister.load_balancers.build(
+        :ems_ref => forwarding_rule.id.to_s,
+        :name    => forwarding_rule.name
+      )
+
+      if forwarding_rule.target
+        # Make sure we link the target link back to this instance for future
+        # back-references
+        @target_pool_link_to_load_balancers[forwarding_rule.target] ||= Set.new
+        @target_pool_link_to_load_balancers[forwarding_rule.target].add(persister_load_balancer)
+      end
+    end
+
+    load_balancer_pools
+
+    forwarding_rules.each do |forwarding_rule|
+      load_balancer_listener(persister.load_balancers.lazy_find(forwarding_rule.id.to_s), forwarding_rule)
+    end
+  end
+
+  def load_balancer_listener(persister_load_balancer, forwarding_rule)
+    # Only TCP/UDP/SCTP forwarding rules have ports
+    has_ports = %w(TCP UDP SCTP).include?(forwarding_rule.ip_protocol)
+    port_range = (parse_port_range(forwarding_rule.port_range) if has_ports)
+
+    persister_load_balancer_listener = persister.load_balancer_listeners.build(
+      :name                     => forwarding_rule.name,
+      :ems_ref                  => forwarding_rule.id.to_s,
+      :load_balancer_protocol   => forwarding_rule.ip_protocol,
+      :instance_protocol        => forwarding_rule.ip_protocol,
+      :load_balancer_port_range => port_range,
+      :instance_port_range      => port_range,
+      :load_balancer            => persister_load_balancer
+    )
+
+    load_balancer_listener_pool(persister_load_balancer_listener, forwarding_rule)
+  end
+
+  def load_balancer_listener_pool(persister_load_balancer_listener, forwarding_rule)
+    persister_lb_listener_pool = persister.load_balancer_listener_pools.build(
+      :load_balancer_listener => persister_load_balancer_listener,
+      :load_balancer_pool     => @target_pool_index[forwarding_rule.target]
+    )
+
+    persister_load_balancer_listener.load_balancer_listener_pools ||= []
+    persister_load_balancer_listener.load_balancer_listener_pools << persister_lb_listener_pool
+  end
+
+  def load_balancer_pools
+    collector.target_pools.each do |target_pool|
+      persister_load_balancer_pool = persister.load_balancer_pools.build(
+        :ems_ref => target_pool.id.to_s,
+        :name    => target_pool.name
+      )
+
+      @target_pool_index[target_pool.self_link] = persister_load_balancer_pool
+
+      load_balancer_pool_members(persister_load_balancer_pool, target_pool)
+      load_balancer_health_check(target_pool)
+    end
+  end
+
+  def load_balancer_pool_members(persister_load_balancer_pool, target_pool)
+    target_pool.instances.to_a.each do |member_link|
+      persister_lb_pool_member = persister.load_balancer_pool_members.find(Digest::MD5.base64digest(member_link))
```

Comment:
@Ladas same question as above, with `lazy_find`, there is no condition when to build `lb_pool_member` and when not

Reply:
lets do this as the volumes, just create this everytime and change specs